File: _2_Palindrome_Number.java


public class _2_Palindrome_Number {
    public static void main(String[] args) {
        Solution sol = new Solution();
        boolean result = sol.isPalindrome(0);
        System.out.println("Result: " + result);
    }
}
// My solution
class Solution {
    public boolean isPalindrome(int x) {

        if ( x < 0) {
            return false;
        }
        
        int reversed = reverseNumber(x);

        return reversed == x;
    }

    public int reverseNumber(int num) {
        int reversed = 0;
        while (num != 0) {
            int digit = num % 10;        
            reversed = reversed * 10 + digit;  
            num = num / 10;             
        }
        return reversed;
    }
}

File: _2_palindromeNumber.ts
function reverseNumber(num: number): number {
    let reversed = 0;
    while (num !== 0) {
        const digit = num % 10;
        reversed = reversed * 10 + digit;
        num = Math.floor(num / 10);
    }
    return reversed;
}

function isPalindrome(x: number): boolean {
    if (x < 0) {
        return false;
    }
    
    const reversed = reverseNumber(x);
    return reversed === x;
}

// Pruebas (esto no va en LeetCode, solo para probar localmente)
console.log("Result:", isPalindrome(1001)); // true
console.log("Result:", isPalindrome(12321)); // true
console.log("Result:", isPalindrome(12345)); // false
